backends/edk2-compat/edk2-svc-validate.c

```diff
@@ -332,13 +334,42 @@ int validateESL(const unsigned char *eslBuf, size_t buflen, const char *key)
 	return SUCCESS;
 }
 
+static int validate_sbat (uint8_t *sbat_data)
+{
+        char *new_line = NULL, *context = NULL;
+        char *data = strdup((char*)sbat_data);
+        int number_of_commas = 0, i =0;
+
+        for (new_line = strtok_r(data, "\n", &context);
+              new_line != NULL;
+              new_line = strtok_r(NULL, "\n", &context)) {
+
+                number_of_commas = 0;
+                i = 0;
+                while (new_line[i] != '\0') {
+                        if (new_line[i] == ',')
+                                number_of_commas++;
+			i++;
+                }
+
+		if (number_of_commas != 1) {
+			free(data);
+			return 0;
+		}
+	}
+
+	free(data);
+
+	return 1;
+}
+
```

Comment:
Looks like this function is a source of a few buffer overreads. This is not a major concern and is due to `strtok_r` and `strdup` reading past the length of `sbat_data` since `\0` is not found in the sbat buffer. It is encouraged that we address this just to ensure that valgrind is happy and future tests can pass.

Just a suggestion:
```suggestion
static int validate_sbat (const uint8_t *sbat_data, size_t sbat_len)
{
	int number_of_commas = 0;

	for (size_t i = 0; i < sbat_len; i++) {
		if (sbat_data[i] == ',')
			number_of_commas++;
		if (sbat_data[i] == '\n') {
			if (number_of_commas != 1)
				return 0;
			number_of_commas = 0;
		}	
	}

	return 1;
}
```